packages/compass-e2e-tests/tests/collection-aggregations-tab.test.ts

```diff
@@ -497,6 +497,153 @@ describe('Collection aggregations tab', function () {
     });
   });
 
+  async function runAggregation() {
+    await browser.clickVisible(Selectors.RunPipelineButton);
+    const resultsWorkspace = await browser.$(
+      Selectors.AggregationResultsWorkspace
+    );
+    await resultsWorkspace.waitForDisplayed();
+  }
+
+  async function editPipeline() {
+    await browser.clickVisible(Selectors.EditPipelineButton);
+    const builderWorkspace = await browser.$(
+      Selectors.AggregationBuilderWorkspace
+    );
+    await builderWorkspace.waitForDisplayed();
+  }
+
+  async function getDocuments() {
+    // Switch to JSON view so it's easier to get document value
+    await browser.clickVisible(
+      Selectors.AggregationResultsJSONListSwitchButton
+    );
+    // Get all visible documents
+    const documents = await browser.$$(Selectors.DocumentJSONEntry);
+    // Get ace editor content and parse it
+    const parsed = await Promise.all(
+      documents.map(async (doc) => {
+        const aceEditor = await doc.$('.ace_content');
+        return JSON.parse(await aceEditor.getText());
+      })
+    );
+    return parsed;
+  }
+
+  it('supports running and editing aggregation', async function () {
+    // Set first stage to match
+    await browser.focusStageOperator(0);
+    await browser.selectStageOperator(0, '$match');
+    await browser.setAceValue(Selectors.stageEditor(0), '{ i: 5 }');
+
+    // Run and wait for results
+    await runAggregation();
+
+    // Get all documents from the current results page
+    const docs = await getDocuments();
+
+    expect(docs).to.have.lengthOf(1);
+    expect(docs[0]).to.have.property('_id');
+    expect(docs[0]).to.have.property('i', 5);
+    expect(docs[0]).to.have.property('j', 0);
+
+    // Go back to the pipeline builder
+    await editPipeline();
+
+    // Change match filter
+    await browser.setAceValue(
+      Selectors.stageEditor(0),
+      '{ i: { $gte: 5, $lte: 10 } }'
+    );
+
+    // Run and wait for results
+    await runAggregation();
+
+    // Get all documents from the current results page
+    const updatedDocs = await getDocuments();
+
+    // Check that the documents are matching pipeline
+    expect(updatedDocs).to.have.lengthOf(6);
+    expect(updatedDocs[0]).to.have.property('i', 5);
+    expect(updatedDocs[1]).to.have.property('i', 6);
+    expect(updatedDocs[2]).to.have.property('i', 7);
+    expect(updatedDocs[3]).to.have.property('i', 8);
+    expect(updatedDocs[4]).to.have.property('i', 9);
+    expect(updatedDocs[5]).to.have.property('i', 10);
+
+    // Go back to the pipeline builder because that's what beforeEach expects
+    await editPipeline();
```

Comment:
One issue with this pattern is that if this test fails, all the next ones will likely fail as well if they expect to start in a specific state. Maybe safer to stick this in the beforeEach?

Reply:
Makes sense, will move!

Reply:
Ah, actually I was wrong to add this in the first place, `beforeEach` already does a full reconnect before every tests so this is not an issue and this call is not needed, I'll just remove it

Reply:
Hmm. Not sure why that reconnects in the beforeEach rather than the before. That's probably overkill and it should be enough to get to the aggregations tab and then click new pipeline to get back to a known state, but I suppose we can deal with that later. ie. if we get a lot of tests in this file it could get pretty slow.

Reply:
Aah I see that happened here https://github.com/mongodb-js/compass/commit/f089dd7c0be61940736afd1e2e8fcbade84b565c

Given that it replaces the db in beforeEach I suppose that's sorta justified. Could maybe just click refresh in sidebar, though.

Reply:
Sounds like a good way of speed it up, yeah, I'll take a look in a separate PR, we don't have a refresh button test at the moment, so might take me some time to figure out what's a good way to do it and be able to wait for the refresh to happen